patches/server/0005-Paper-config-files.patch

```diff
@@ -725,6 +725,7 @@ index 0000000000000000000000000000000000000000..45f1436cdd4b81b621ab71e4336c2aa6
 +        public boolean useAlternativeLuckFormula = false;
 +        public boolean useDimensionTypeForCustomSpawners = false;
 +        public boolean strictAdvancementDimensionCheck = false;
++        public int compressionLevel = -1;
```

Comment:
Would probably be better to use IntOrDefault here.
-1 yields default compression for the zip deflater as well as for the native ones velocity implements (6 iirc)

Reply:
Ok so I have tried this and I don't seem to be able to do that.
It tells me that `No TypeSerializer found for field compressionLevel of type class io.papermc.paper.configuration.type.IntOr$Default`, before it just did not want to create the `misc` section at all.
To my eyes I have made everything like where it is already in use. 
The updated `GlobalConfiguration` variable:
```java
public IntOr.Default compressionLevel = IntOr.Default.USE_DEFAULT
```
The updated line in `Connection#setupCompression()`:
```java
[...].create(io.papermc.paper.configuration.GlobalConfiguration.get().misc.compressionLevel.or(-1));
```
Do I have to change anything else besides that?

Reply:
Ah so i think this is why:
https://github.com/PaperMC/Paper/blob/master/patches/server/0005-Paper-config-files.patch#L1166
(only the world-config has the IntOr Serializers registered)
What are we doing now? Should we create a separate PR adding the stuff to the global config or should I include that here?

Reply:
include the registration here :+1:

Reply:
should be fixed now.